src/main/java/upeldev/com/github/upel3/services/ActivityService.java

```diff
@@ -0,0 +1,56 @@
+package upeldev.com.github.upel3.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+import upeldev.com.github.upel3.model.*;
+import upeldev.com.github.upel3.repositories.ActivityRepository;
+
+import java.util.List;
+
+@Service
+public class ActivityService {
+    private final ActivityRepository activityRepository;
+
+    @Autowired
+    public ActivityService(ActivityRepository activityRepository) {
+        this.activityRepository = activityRepository;
+    }
+
+    public Activity findActivityById(Long id){
+        return activityRepository.findActivityById(id);
+    }
+
+    public List<Activity> findActivityByCourse(Course course){
+        return activityRepository.findActivityByCourse(course);
+    }
+
+    public boolean canUserAddActivity(Activity activity, User user){
+        if(user.getRoles().contains(Role.ADMIN) || user.getRoles().contains(Role.LECTURER)){
+            User lecturer = activity.getCourse().getLecturer();
```

Comment:
What if there are more than one lecturer should be assigned to the course?

Reply:
We can leave it for now and I'll adjust it while working on the course model